include/nlohmann/detail/output/binary_writer.hpp

```diff
@@ -847,11 +847,11 @@ class binary_writer
                     oa->write_character(to_char_type('['));
                 }
 
-                if (use_type && !j.m_data.m_value.binary->empty())
+                if (use_type && (use_bjdata || !j.m_data.m_value.binary->empty()))
```

Comment:
@nlohmann 
And that specific test-case you commented on has changed due to this line. In order to ensure that encoding an empty `json::binary` still decodes to `json::binary` I force the insertion of the type here regardless of whether there are contents or not.

That said this problem will still exist if other libraries are used (e.g. python encodings an empty `bytearray` and doesn't include the type).